src/NumericalAlgorithms/SphericalHarmonics/YlmSpherepackHelper.cpp

```diff
@@ -167,4 +167,13 @@ void MemoryPool::free(const gsl::not_null<double*> to_be_freed) {
   ERROR("Attempt to free temp that was never allocated.");
 }
 
+void MemoryPool::clear() {
+  for (auto& elem : memory_pool_) {
+    if(elem.currently_in_use) {
```

Comment:
Is this `clear()` function really not being tested? If so, could you add (or extend) the tests to cover it?

Reply:
Ok, I did that and even more.   My new commit changes all the error tests in Test_YlmSpherepack into CHECK_THROWS_WITH.